install_sct.bat

```diff
@@ -79,7 +79,9 @@ rem Allow user to set a custom installation directory
 :while_loop_sct_dir
   echo:
   echo ### SCT will be installed here: [%SCT_DIR%]
-  set keep_default_path=yes

```

Comment:
[Oops!](https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/15761025498/job/44427528943?pr=4935#step:3:24) It looks like this default value of `yes` may have been what allowed the script to run in non-interactive mode.

I'll have to rethink the flow of logic to allow for non-interactive mode.

Reply:
The fix here is quite simple! If the `keep_default_path` value doesn't change from the `SET /P` command, then we can assume it's a noninteractive shell and auto-accept the default.

Likewise, we can apply the same logic to the `new_install` variable to prevent an infinite loop there, as well (in case the default install location is invalid and we're unable to update it).